sdk/analytics/src/main/java/com/klaviyo/analytics/networking/requests/KlaviyoApiRequest.kt

```diff
@@ -149,7 +80,12 @@ internal open class KlaviyoApiRequest(
             val time = json.getLong(TIME_JSON_KEY)
             val uuid = json.getString(UUID_JSON_KEY)
 
-            return KlaviyoApiRequest(urlPath, method, time, uuid).apply {
+            return when (json.optString(TYPE_JSON_KEY)) {
```

Comment:
Using simple class name map to decode JSON payload back to the original subtype class

Reply:
So was the problem that before you were instantiating a `KlaviyoApiRequest` instead of mapping back to what it was originally from the json provided?

Reply:
Cause the logic here makes sense to me but I think the use of concrete classes within the parent class is a bit of an anti-pattern.

Reply:
I'm wondering if it would make sense to move this logic up a bit higher into the caller (`KlaviyoApiClient`) in this case, to determine the class to instantiate from the given JSON before any request objects are invoked.

Reply:
Ya I'd heavily prefer if we didn't have an abstract/parent class dependent on its children. Doing it dynamically as well also sounds fine. I'd just strive to keep these references out of here so do away with the "circular dependency" we got going on here.

Reply:
good call, definitely felt a little dirty doing it this way

Reply:
Tried dynamic, but it requires reflection and just felt wrong. Thinking now the most straightforward would be a helper object like `object KlaviyoApiRequestSerializer` so that this isn't directly in the `KlaviyoApiRequest` class. I liked putting it in `KlaviyoApiClient` at first, but it makes some dependency issues for unit tests